exchanges/exchange_test.go

```diff
@@ -2230,7 +2230,15 @@ func TestSetAPIURL(t *testing.T) {
 	mappy.Mappymap["RestSpotURL"] = "hi"
 	b.API.Endpoints = b.NewEndpoints()
 	err = b.SetAPIURL()
-	if err != nil {
-		t.Error(err)
+	if err == nil {
+		t.Error("expecting an error since invalid url value is input")
+	}
+	mappy.Mappymap = make(map[string]string)
+	b.Config.API.Endpoints = mappy.Mappymap
+	mappy.Mappymap["RestSpotURL"] = "www.google.com"
+	b.API.Endpoints = b.NewEndpoints()
+	err = b.SetAPIURL()
+	if err == nil {
+		t.Error("expecting an error since invalid url value is input")
```

Comment:
Thanks for doing that. Now if you could just add coverage to a valid scenario to get the rest of the functions covered, then all good!